test/jdk/sun/tools/jhsdb/JStackStressTest.java

```diff
@@ -0,0 +1,115 @@
+ * @requires vm.hasSA
+ * @library /test/lib
+ * @run main/timeout=240 JStackStressTest
+ */
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+import jdk.test.lib.JDKToolLauncher;
+import jdk.test.lib.JDKToolFinder;
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.process.ProcessTools;
+import jdk.test.lib.SA.SATestUtils;
+import jdk.test.lib.Utils;
+
+public class JStackStressTest {
+
+    static Process jShellProcess;
+
+    public static void testjstack() throws IOException {
+        launchJshell();
+        long jShellPID = jShellProcess.pid();
+        OutputAnalyzer jshellOutput = new OutputAnalyzer(jShellProcess);
+
+        try {
+            // Do 4 jstacks on the jshell process as it starts up
+            for (int i = 1; i <= 4; i++) {
+                JDKToolLauncher launcher = JDKToolLauncher.createUsingTestJDK("jhsdb");
+                launcher.addVMArgs(Utils.getTestJavaOpts());
+                launcher.addToolArg("jstack");
+                launcher.addToolArg("--pid=" + Long.toString(jShellPID));
+
+                System.out.println("###### Starting jstack iteration " + i + " against " + jShellPID);
+                long startTime = System.currentTimeMillis();
+                ProcessBuilder processBuilder = SATestUtils.createProcessBuilder(launcher);
+                OutputAnalyzer output = ProcessTools.executeProcess(processBuilder);
+                System.out.println("jhsdb jstack stdout:");
+                System.out.println(output.getStdout());
+                System.out.println("jhsdb jstack stderr:");
+                System.out.println(output.getStderr());
+                long elapsedTime = System.currentTimeMillis() - startTime;
+                System.out.println("###### End of all output for iteration " + i +
+                                   " which took " + elapsedTime + "ms");
+                output.shouldHaveExitValue(0);
+                // This will detect most SA failures, including during the attach.
+                output.shouldNotMatch("^sun.jvm.hotspot.debugger.DebuggerException:.*$");
+                // This will detect unexpected exceptions, like NPEs and asserts, that are caught
+                // by sun.jvm.hotspot.tools.Tool.execute().
+                output.shouldNotMatch("^Error: .*$");
+            }
+        } catch (Exception ex) {
+            throw new RuntimeException("Test ERROR " + ex, ex);
+        } finally {
+            try (OutputStream out = jShellProcess.getOutputStream()) {
+                out.write("/exit\n".getBytes());
+                out.flush();
+            }
+            System.out.println("jshell Output: " + jshellOutput.getOutput());
+            if (jShellProcess.isAlive()) {
+                System.out.println("Destroying jshell");
+                jShellProcess.destroy();
+                System.out.println("Jshell destroyed");
+            } else {
+                System.out.println("Jshell not alive");
+            }
```

Comment:
Shouldn't we use `Process::waitFor` at this point to ensure the jshell is terminated? This test might not have enough time to shutdown since issuing `/exit`.
`destroy()` should be called when timeout is happened at `waitFor()`.

Reply:
Yeah, I'll do that. Not sure why I didn't do it that way in the first place. I think I copied this from somewhere, but now I can't find where.

Reply:
Ah, it was JShellHeapDumpTest.java, which does the same thing (it's in a different test directory, so my grep missed it). There really is no reason not to just forcibly kill it, since its job is done and we don't really need its output or a graceful exit. However, for some reason I partially decided to try to get it to exit gracefully by adding the /exit command, and then get the output. So I'll still add the waitfor.